### PR TITLE
Fix generate_update_query when there is no where_clause

### PR DESCRIPTION
**Pull request by cmackinnon-cedar**
## Summary

`generate_update_query` has a bug where the query ends up being `WHERE () AND ...` (invalid SQL) in several cases where where_clause would end up empty.
This change makes no functional changes to the queries at all, but allows for use of update queries in cases where we do not need conditions on update. 

##### How can reviewers test your change manually?
Unit tests

##### What automated tests did you add? If none, please state why.
Unit tests

## Security Impact
Please answer the questions below about your PR. 
If yes, please explain and add `cedar-team/security` as reviewers.

##### Add/modify authentication, authorization, encryption, or HTTP headers?
No
##### Add/modify any PII or PHI data fields?
No
##### Add new third parties the app interacts with (e.g., Nordis/Stripe)?
No
##### Ask patients to enter a new kind of data, (e.g. insurance capture)?
No
##### Impact any of the other [Security Engineering Practices]
No

### CODE REVIEW

**Review by bfinnell-cedar — DISMISSED** [review]
- [ ] I fully understand what the code is doing and its context
- [ ] The code is the best way to accomplish this problem (I would code it similarly)
- [ ] The code is well organized, easy to understand and matches Cedar design patterns and best practices
- [ ] The code handles edge cases (that I can think of) and has tests to validate them
- [ ] The appropriate people are reviewing (knowledge and security)
- [ ] I tested the code personally (Not required)

File: django_bulk_load/queries.py [hunk 1]
```diff
@@ -290,15 +290,28 @@ def generate_update_query(
             else:
                 where_clause = distinct_null_clause
 
+    # Check if where_clause is actually empty by checking if compare_fields was empty
+    # and no update_if_null_fields were provided
+    has_where_conditions = (
+        (compare_fields and not update_where) or 
+        update_if_null_fields or 
+        update_where
```
Comment (jwu-cedar) [inline django_bulk_load/queries.py:298]: Alternatively can we check if the where_clause equals "WHERE ()" and decide to skip? I'm not familiar enough with this code to be sure that this covers all the cases in which the WHERE clause would be empty.

**Review by jwu-cedar — APPROVED** [review]
LGTM!
- [ ] I fully understand what the code is doing and its context
- [ ] The code is the best way to accomplish this problem (I would code it similarly)
- [ ] The code is well organized, easy to understand and matches Cedar design patterns and best practices
- [ ] The code handles edge cases (that I can think of) and has tests to validate them
- [ ] The appropriate people are reviewing (knowledge and security)
- [ ] I tested the code personally (Not required)
